packages/test-worker/src/test-worker-instance.ts

```diff
@@ -91,16 +92,44 @@ export class TestWorkerInstance implements ITestWorkerInstance {
     ) {
         this.worker = await this.getWorker();
 
-        // Calling external hooks to compile source
-        const compiledSource = await this.compileSource(file.content, file.path);
-        // TODO implement code instrumentation here
+        let compiledFile;
 
-        const compiledFile = {
-            path: file.path,
-            content: compiledSource
-        };
+        const initialFilePaths = [ file.path ];
+
+        const filePaths = await this.beforeCompile(initialFilePaths);
+
+        const files = await Promise.all(
+            filePaths.map(async (path) => {
```

Comment:
probably introduce throttle or queue? possible high load to FS in case of a lot of files